src/interpreter/ByteCode.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Walrus.h"
+
+#include "interpreter/ByteCode.h"
+
+namespace Walrus {
+
+size_t ByteCode::getSize()
+{
+    OpcodeKind opcode = this->opcode();
+
+    ASSERT(g_byteCodeInfo[opcode].m_size > 0);
+
+    switch (opcode) {
+    case EndOpcode: {
+#if !defined(NDEBUG)
+        return g_byteCodeInfo[EndOpcode].m_size + sizeof(ByteCodeStackOffset) * static_cast<End*>(this)->functionType()->result().size();
+#else
+        return g_byteCodeInfo[EndOpcode].m_size;
```

Comment:
The problem is that this is incorrect in release mode. Optimizations will not work.